Source/WebCore/html/FileInputType.cpp

```diff
@@ -276,32 +224,43 @@ void FileInputType::createShadowSubtree()
     ASSERT(element());
     ASSERT(element()->shadowRoot());
 
-    auto button = element()->multiple() ? UploadButtonElement::createForMultiple(element()->document()) : UploadButtonElement::create(element()->document());
+    auto button = HTMLInputElement::create(inputTag, element()->document(), nullptr, false);
+    {
+        ScriptDisallowedScope::EventAllowedScope eventAllowedScopeBeforeAppend { button };
+        button->setType(InputTypeNames::button());
+        button->setPseudo(ShadowPseudoIds::fileSelectorButton());
+        button->setValue(element()->multiple() ? fileButtonChooseMultipleFilesLabel() : fileButtonChooseFileLabel());
+    }
     ScriptDisallowedScope::EventAllowedScope eventAllowedScope { *element()->userAgentShadowRoot() };
     element()->userAgentShadowRoot()->appendChild(ContainerNode::ChildChange::Source::Parser, button);
     disabledStateChanged();
 }
 
+static RefPtr<HTMLInputElement> fileSelectorButton(RefPtr<Element> element)
```

Comment:
```suggestion
static RefPtr<HTMLInputElement> fileSelectorButton(const Element& element)
```

Instead of asserting the presence of element, you could just take in a reference. (Also you shouldn't pass in a smart pointer here, the caller is supposed to protect the element in this case and we're also just calling trivial getters on element)